File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> m=new HashMap<>();
        for(int i=0;i<strs.length;i++){
            char[] p=strs[i].toCharArray();
            Arrays.sort(p);
            String x=new String(p);
            if(!m.containsKey(x)){
                m.put(x,new ArrayList());
                //m.get(x).add(strs[i]);
            }
            m.get(x).add(strs[i]);
        }
        List<List<String>> ans=new ArrayList<List<String>>();
        ans.addAll(m.values());
        return ans;
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        d={}
        l=[[]]
        for i in strs:
            s=str(sorted(i))
            if s in d.keys():
                d[s].append(i)
            else:
                x=list()
                x.append(i)
                d[s]=x
        return d.values()